packages/cloud_functions/cloud_functions/android/src/main/java/io/flutter/plugins/firebase/functions/FlutterFirebaseFunctionsPlugin.java

```diff
@@ -55,6 +62,14 @@ public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
     channel = null;
   }
 
+  private void registerEventChannel(Map<String, Object> arguments) {
+    final String eventId = (String) Objects.requireNonNull(arguments.get("eventChannelId"));
+    final String eventChannelName = METHOD_CHANNEL_NAME + "/" + eventId;
+    final EventChannel eventChannel =
+        new EventChannel(pluginBinding.getBinaryMessenger(), eventChannelName);
+    eventChannel.setStreamHandler(this);
```

Comment:
I would rather have another class that implements EventChannel.StreamHandler so it's properly separated and easier to review

Reply:
Okay. I'll update the PR